### ISSUE TITLE
Thread 'tokio-runtime-worker' panicked at 'header is last in set and contains standard change signal

### ISSUE THREAD

**Issue by stakeworks** [litentry/litentry-parachain#1481]
```
This morning node crashed a couple of times (see image below) with this error:

2023-03-17 06:06:06 [Relaychain] <E2><9C><A8> Imported #14680211 (0xaafb<E2><80><A6>559f)
2023-03-17 06:06:06 [Parachain] Starting collation. relay_parent=0xaafbd843dcd45d8bf26788541f39206886a93e967cf45dc26b8a8404b93e559f at=0x7b175d11eb1dde621dc7c14362acda3a451491e1227a6691068ae82101e38019
2023-03-17 06:06:06 [Parachain] <E2><9C><A8> Imported #1897321 (0x0d5f<E2><80><A6>ec8e)
2023-03-17 06:06:08 [Relaychain] <F0><9F><92><A4> Idle (41 peers), best: #14680211 (0xaafb<E2><80><A6>559f), finalized #14680208 (0xd015<E2><80><A6>f8d4), <E2><AC><87> 293.5kiB/s <E2><AC><86> 390.6kiB/s
2023-03-17 06:06:08 [Parachain] <F0><9F><92><A4> Idle (15 peers), best: #1897320 (0x7b17<E2><80><A6>8019), finalized #1897318 (0xba05<E2><80><A6>0d88), <E2><AC><87> 8.5kiB/s <E2><AC><86> 7.9kiB/s
====================
Version: 0.9.15-unknown
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
Thread 'tokio-runtime-worker' panicked at 'header is last in set and contains standard change signal; must have justification; qed.', /usr/local/cargo/git/checkouts/substrate-7e08433d4c370a21/f38bd66/client/finality-grandpa/src/warp_proof.rs:135
This is a bug. Please report it at:
        https://github.com/litentry/litentry-parachain/issues/new
litentry-collator.service: Main process exited, code=exited, status=1/FAILURE
litentry-collator.service: Failed with result 'exit-code'.
litentry-collator.service: Consumed 2min 34.908s CPU time.
litentry-collator.service: Scheduled restart job, restart counter is at 8.
Stopped litentry-collator.service.
litentry-collator.service: Consumed 2min 34.908s CPU time.
Started litentry-collator.service.
2023-03-17 06:08:11 Litentry node
2023-03-17 06:08:11 <E2><9C><8C><EF><B8><8F>  version 0.9.15-unknown
2023-03-17 06:08:11 <E2><9D><A4><EF><B8><8F>  by Litentry Dev, 2017-2023
2023-03-17 06:08:11 <F0><9F><93><8B> Chain specification: Litentry
2023-03-17 06:08:11 <F0><9F><8F><B7>  Node name: StakeWorks | Litentry | MEV02
2023-03-17 06:08:11 <F0><9F><91><A4> Role: AUTHORITY
2023-03-17 06:08:11 <F0><9F><92><BE> Database: RocksDb at /home/lit/.litentry-collator/chains/litentry/db/full
2023-03-17 06:08:11 <E2><9B><93>  Native runtime: litentry-parachain-9150 (litentry-parachain-0.tx1.au1)
2023-03-17 06:08:16 Parachain id: Id(2013)
```

Operating system: linux
CPU architecture: x86_64
Target environment: gnu
CPU: AMD Ryzen 7 3700X 8-Core Processor
CPU cores: 8
Memory: 64259MB
Kernel: 5.15.0-25-generic
Linux distribution: Ubuntu 22.04 LTS
Virtual machine: no

![image](https://user-images.githubusercontent.com/19406838/225867534-499fa40e-7620-47b8-8e74-ad30c08a9a79.png)

**Comment by Kailai-Wang** [comment]
Thanks for the report.

This is also the first time that we have seen this. The stack trace is stripped so of little use, unfortunately :( The hardware setup is more than enough. 

Has it been stable now? I hope it's just some random subtle bug that is self-recovered. We'll keep an eye on this though.

**Comment by stakeworks** [comment]
Hello Kai,

 

Yes, looks stable now. Will monitor it closely over the next few days.

 



 

Van: Kai ***@***.***> 
Verzonden: vrijdag 17 maart 2023 18:14
Aan: litentry/litentry-parachain ***@***.***>
CC: WebDev | Stake.Works ***@***.***>; Author ***@***.***>
Onderwerp: Re: [litentry/litentry-parachain] Thread 'tokio-runtime-worker' panicked at 'header is last in set and contains standard change signal (Issue #1481)

 

Thanks for the report.

This is also the first time that we have seen this. The stack trace is stripped so of little use, unfortunately :( The hardware setup is more than enough.

Has it been stable now? I hope it's just some random subtle bug that is self-recovered. We'll keep an eye on this though.

**Comment by stakeworks** [comment]
Node crashed this evening with the same error:

![image](https://user-images.githubusercontent.com/19406838/226143297-d03065c0-a378-47cd-8cd3-d0510bf71073.png)

**Comment by Kailai-Wang** [comment]
Hi @wwwAndries how is it in the last few days? 

We have been focusing on the product launch recently so we don't have much time to look into this, but we are working on the latest polkadot-v0.9.39 integration and hope it will get better

**Comment by stakeworks** [comment]
Hello Kai,

 

Last two days went fine, no problems. When will the new version be released?

 

Van: Kai ***@***.***> 
Verzonden: zaterdag 25 maart 2023 21:40
Aan: litentry/litentry-parachain ***@***.***>
CC: WebDev | Stake.Works ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [litentry/litentry-parachain] Thread 'tokio-runtime-worker' panicked at 'header is last in set and contains standard change signal (Issue #1481)

 

Hi @wwwAndries <https://github.com/wwwAndries>  how is it in the last few days?

We have been focusing on the product launch recently so we don't have much time to look into this, but we are working on the latest polkadot-v0.9.39 integration and hope it will get better

**Comment by Kailai-Wang** [comment]
Great to hear that. We'll try to release the new client this week, based on `polkadot-v0.9.39`

**Comment by Kailai-Wang** [comment]
> Great to hear that. We'll try to release the new client this week, based on `polkadot-v0.9.39`

There have been some delay in delivering the new version -- we are still working on it (mainly the TEE part)

**Comment by Kailai-Wang** [comment]
We have released `v0.9.16` which is based on polkadot-v0.9.39 - if the problem hasn't occurred for a while, we'd like to close this issue.

**Comment by Kailai-Wang** [comment]
I'm closing this ticket now, please feel free to reopen it or create a new issue once the problem happens again.
